gateway/ln-gateway/src/lib.rs

```diff
@@ -273,8 +273,14 @@ impl LnGateway {
         // We use a random federation as the default (works because we only have one federation registered)
         //
         // TODO: Use subscribe intercept htlc streams to avoid actor selection with every intercepted htlc!
-        self.actors.lock().await.values().collect::<Vec<_>>()[0]
-            .buy_preimage_internal(&payment_hash, &invoice_amount)
+        let lock = &self.actors.lock().await;
```

Comment:
I don't know. I just extended the lifetime of a guard in here, the locking was already in place.

Reply:
@okjodom?

Reply:
lock because you are selecting an actor to use. actors is a mutex because we can create and reference new actors, or remove existing actors at runtime

PS: Actor selection and TODO described in line 275 should be complete very soon! I'm okay with the lock for now